sources/models/src/modeled_types/oci_defaults.rs

```diff
@@ -0,0 +1,145 @@
+use scalar_derive::Scalar;
+use serde::{Deserialize, Serialize};
+use serde_plain::{derive_display_from_serialize, derive_fromstr_from_deserialize};
+
+/// OciDefaultsCapability specifies which process capabilities are
+/// allowed to be modified in Bottlerocket. The Linux process
+/// capabilities are defined in the Linux manpages (see the man7.org link below).
+/// Default values can be found in: sources/models/shared-defaults/oci-capabilities.toml
+/// OCI spec: https://github.com/opencontainers/runtime-spec/blob/main/config.md#linux-process
+/// Linux kernel capabilities: https://man7.org/linux/man-pages/man7/capabilities.7.html
+///
+/// Default capabilities are specified in the following file:
+/// * sources/models/shared-defaults/oci-capabilities.toml
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash, Scalar, Serialize, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub enum OciDefaultsCapability {
+    AuditControl,
+    AuditRead,
+    AuditWrite,
+    BlockSuspend,
+    Bpf,
+    CheckpointRestore,
+    Chown,
+    DacOverride,
+    DacReadSearch,
+    Fowner,
+    Fsetid,
+    IpcLock,
+    IpcOwner,
+    Kill,
+    Lease,
+    LinuxImmutable,
+    MacAdmin,
+    MacOverride,
+    Mknod,
+    NetAdmin,
+    NetBindService,
+    NetBroadcast,
+    NetRaw,
+    Perfmon,
+    Setgid,
+    Setfcap,
+    Setpcap,
+    Setuid,
+    SysAdmin,
+    SysBoot,
+    SysChroot,
+    SysModule,
+    SysNice,
+    SysPacct,
+    SysPtrace,
+    SysRawio,
+    SysResource,
+    SysTime,
+    SysTtyConfig,
+    Syslog,
+    WakeAlarm,
+}
+
+impl OciDefaultsCapability {
+    /// Converts from Bottlerocket's kabob-case name into the Linux capability name, e.g. turns
+    /// `wake-alarm` into `CAP_WAKE_ALARM`.
+    pub fn to_linux_string(&self) -> String {
+        format!("CAP_{}", self.to_string().to_uppercase().replace('-', "_"))
+    }
+}
+
+#[cfg(test)]
+mod oci_defaults_capabilities {
+    use super::*;
+
+    fn check_capability_strings(cap: OciDefaultsCapability, bottlerocket: &str, linux: &str) {
+        let actual_bottlerocket = cap.to_string();
+        let actual_linux = cap.to_linux_string();
+        assert_eq!(bottlerocket, actual_bottlerocket);
+        assert_eq!(linux, actual_linux);
+    }
+
+    #[test]
+    fn linux_capability_strings() {
+        check_capability_strings(
+            OciDefaultsCapability::AuditControl,
+            "audit-control",
+            "CAP_AUDIT_CONTROL",
+        );
```

Comment:
These small checks are fine, but ideally we can make this unit test test for all possible enum variants so as that list potentially changes, we're always covered. A quick look on how to do that led me to https://docs.rs/strum_macros/0.24.3/strum_macros/derive.EnumIter.html. The `strum` crate seems lightweight (5.64kb) enough so I don't mind adding it as a test dependency.

Same comment for the link above `OciDefaultsCapability`.

Reply:
Ah silly me, I just realized what's being validated here. No need to do anything here.